File: test/cases/record/test_record_while_loop.pas
program test;
type
  point = record
    x, y: integer
  end;
var
  p: point;

begin
  p.x := 0;
  p.y := 0;

  while p.x < 5 do
  begin
    p.x := p.x + 1;
    p.y := p.y + 2;
    write(p.x);
    write(p.y)
  end
end.

File: test/cases/record/test_record_while_loop.c
#include <stdio.h>
typedef struct {
    int x;
    int y;
} point;
point p;


int main(void) {
    p.x = 0;
    p.y = 0;
    while ((p.x < 5)) {
        p.x = (p.x + 1);
        p.y = (p.y + 2);
        printf("%d", p.x);
        printf("%d", p.y);
    }

    return 0;
}
